homeassistant/components/met/weather.py

```diff
@@ -62,99 +58,77 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     if config.get(CONF_LATITUDE) is None:
         config[CONF_TRACK_HOME] = True
 
-    async_add_entities([MetWeather(config, hass.config.units.is_metric)])
+    await async_setup(hass, config, async_add_entities)
```

Comment:
this should defer to config flow to process, you tben add and import step in config flow.

```suggestion
    hass.async_create_task(
        hass.config_entries.flow.async_init(
            DOMAIN, context={"source": SOURCE_IMPORT}, data=config
        )
    )
```

Reply:
might be better left for future PR though. have a few concerns with the combined usage of async_setup

Reply:
your config flow will need a method like this for import to work:

https://github.com/home-assistant/core/blob/dev/homeassistant/components/roku/config_flow.py#L62

Reply:
Going forward, I will need to create two entities from one configuration - daily and hourly forecast. How will that work when I remove adding the entities? 
That is the whole purpose adding the update coordinator - I made this PR earlier, initially with an option which type you want, then I was advised to create both, and then that it makes redundant API calls and I should add the coordinator. So I need to make sure it will support the initial purpose for doing all this.

Reply:
basically the import step defers to the user input so its migrating the old config into its own config entry which will run through async_setup_entry juat like a user configured from UI